src/clients/Deluge.ts

```diff
@@ -116,9 +134,7 @@ export default class Deluge implements TorrentClient {
 	 * parses the set-cookie header and updates stored value
 	 */
 	private handleResponseHeaders(headers: Headers) {
-		if (headers.has("Set-Cookie")) {
-			this.delugeCookie = headers.get("Set-Cookie").split(";")[0];
-		}
+		this.delugeCookie = headers.get("Set-Cookie")?.split(";")[0] ?? null;
```

Comment:
not sure I agree with this change. just because we don't receive a set cookie doesn't mean we should throw our cookie away.

Reply:
Good point, will update